StarRocks/starrocks
# [Enhancement] Imporve or predicate statistics check

## What type of PR is this：

## Which issues of this PR fixes ：

Fixes #

core modify:
1. update PredicateStatisticsCalculator
   a. update DisConsecutiveOr check
   b. Fix or statistics compute bug(line: 292)

other:
1. update NormalizePredicateRule.inPredicate rule, only extract variable column
2. update planner profile, add level and order, support repeat use
3. rename PredicateStatisticsCalculatingVisitor 
4. remove unuse code

## Problem Summary(Required) ：


2000 predicate query cost 40s -> 8s, other bottlenecks:
1. Expr hashCode and equals, repeat a lot of calculations (1.8s)
2. NormalizePredicateRule.compound, repeat a lot of calculations(3s)
3. StackOverflow

it's hard to modify

```
    1691ms|-- Parser[1] 436ms
    2129ms|-- Planner[1] 7652ms
    2134ms|    -- Analyzer[1] 35ms
    2179ms|    -- Transformer[1] 4081ms
    2225ms|        -- filter1[1] 4023ms
    2225ms|            -- translate[1] 3589ms
    2238ms|                -- HashEquals[16005] 1749ms
    4010ms|                -- rewritePredicate[2] 1805ms
    5815ms|            -- rewriteSubquery[1] 433ms
    6261ms|    -- Optimizer[1] 3453ms
    6358ms|        -- Optimizer.RuleBaseOptimize[1] 3202ms
    6363ms|            -- PushDownPredicate[1] 3049ms
    6928ms|                -- RangeRewriter[1] 693ms
    7621ms|                -- rewritePredicate[1] 1768ms
    9566ms|        -- Optimizer.CostBaseOptimize[1] 127ms
    9582ms|            -- predicate statistics[2] 90ms
    9695ms|        -- Optimizer.PhysicalRewrite[1] 19ms
    9715ms|    -- ExecPlanBuild[1] 67ms
    9782ms|-- Test View[1] 6113ms
```

## Checklist:

- [ ] This pr will affect users' behaviors

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2
